transform/error.go

```diff
@@ -0,0 +1,44 @@
+package transform
+
+import "encoding/json"
+
+const (
+	PluginInvalidInputError = "PluginInvalidInputError"
+	PluginRunError          = "PluginRunError"
+	PluginInvalidIOError    = "PluginInvalidIOError"
```

Comment:
probably `PluginIOError`

Reply:
should this be in a transform/error package? 

Just wondering? I don't think I have strong opinions ATM

Reply:
> should this be in a transform/error package?

yes, I went back and forth but it wasn't big enough for its own package. Might make it seem better if we have it in the separate error package.